### PR TITLE
[SPARK-48775][SQL][STS] Replace SQLContext with SparkSession in STS

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Remove the exposed `SQLContext` which was added in SPARK-46575. And migrate STS internal used `SQLContext` to `SparkSession`.

### Why are the changes needed?

`SQLContext` is not recommended since Spark 2.0, the suggested replacement is `SparkSession`. We should avoid exposing the deprecated class to Developer API in new versions.

### Does this PR introduce _any_ user-facing change?

No. It touched the Developer API added in SPARK-46575, but is not released yet.

### How was this patch tested?

Pass GHA, and `dev/mima` (not breaking changes involved)

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @HyukjinKwon @yaooqinn @LuciferYang

**Comment by LuciferYang** [comment]
also cc @dongjoon-hyun

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIService.scala [hunk 1]
```diff
@@ -103,7 +103,7 @@ private[hive] class SparkSQLCLIService(hiveServer: HiveServer2, sqlContext: SQLC
     getInfoType match {
       case GetInfoType.CLI_SERVER_NAME => new GetInfoValue("Spark SQL")
       case GetInfoType.CLI_DBMS_NAME => new GetInfoValue("Spark SQL")
-      case GetInfoType.CLI_DBMS_VER => new GetInfoValue(sqlContext.sparkContext.version)
+      case GetInfoType.CLI_DBMS_VER => new GetInfoValue(sparkSession.sparkContext.version)
```
Comment (LuciferYang) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIService.scala:106]:
```suggestion
      case GetInfoType.CLI_DBMS_VER => new GetInfoValue(sparkSession.version)
```

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala [hunk 2]
```diff
@@ -49,30 +49,32 @@ object HiveThriftServer2 extends Logging {
...
-   * @param sqlContext SQLContext to use for the server
+   * @param sparkSession SparkSession to use for the server
    * @param exitOnError Whether to exit the JVM if HiveThriftServer2 fails to initialize. When true,
    *                    the call logs the error and exits the JVM with exit code -1. When false, the
    *                    call throws an exception instead.
    */
   @Since("4.0.0")
   @DeveloperApi
-  def startWithContext(sqlContext: SQLContext, exitOnError: Boolean): HiveThriftServer2 = {
+  def startWithSparkSession(
```
Comment (LuciferYang) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala:61]: It seems that aside from the unreleased API added in 4.0, this pr has not caused any breaking changes.

If so, the changes is fine to me. WDYT? @HyukjinKwon  @yaooqinn

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala [hunk 3]
```diff
@@ -82,10 +84,11 @@ object HiveThriftServer2 extends Logging {
    *
    * @param sqlContext SQLContext to use for the server
    */
+  @deprecated("use startWithSparkSession instead", since = "4.0.0")
```
Comment (dongjoon-hyun) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala:87]: nit. `use` -> `Use`

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala (hunk 3 above)
Comment (pan3793) [reply sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala]: thanks, updated.

**Review by dongjoon-hyun — APPROVED** [review]
Thank you for updating. Pending CIs.

**Pull request closed without merging**